substrate/client/network/src/transport.rs

```diff
@@ -57,6 +57,7 @@ pub fn build_transport(
 		// Main transport: DNS(TCP)
 		let tcp_config = tcp::Config::new().nodelay(true);
 		let tcp_trans = tcp::tokio::Transport::new(tcp_config.clone());
+		#[allow(deprecated)]
 		let dns_init = dns::TokioDnsConfig::system(tcp_trans);
```

Comment:
Could we use `libp2p::dns::tokio::Transport::system` instead?

Reply:
`libp2p::dns::tokio::Transport::system` is not yet there in libp2p v0.52.3 this PR is upgrading to. Also, no need for `#[allow(deprecated)]` yet.